traffic_control/clients/python/to_access/__init__.py

```diff
@@ -0,0 +1,620 @@
+	                    action="store_true",
+	                    help=("Also output HTTP request/response lines and headers, and request payload. "
+	                         "This is equivalent to using '--request_headers', '--response_headers' "
+	                         "and '--request_payload' at the same time."))
+	parser.add_argument("--request_headers",
+	                    action="store_true",
+	                    help="Output request method line and headers")
+	parser.add_argument("--response_headers",
+	                    action="store_true",
+	                    help="Output response status line and headers")
+	parser.add_argument("--request_payload",
+	                    action="store_true",
+	                    help="Output request payload (will try to pretty-print if '--pretty' is given)")
+	parser.add_argument("-r", "--raw_path",
+	                    action="store_true",
+	                    help="Request exactly PATH; it won't be prefaced with '/api/{{api_version}}/")
+	parser.add_argument("-a", "--api_version",
+	                    type=float,
+	                    default=1.3,
+	                    help="Specify the API version to request against")
+	parser.add_argument("-p", "--pretty",
+	                    action="store_true",
+	                    help=("Pretty-print payloads as JSON. "
+	                         "Note that this will make Content-Type headers \"wrong\", in general"))
+	parser.add_argument("PATH", help="The path to the resource being requested - omit '/api/1.x'")
+	parser.add_argument("DATA",
+	                    help=("An optional data string to pass with the request. If this is a "
+	                         "filename, the contents of the file will be sent instead."),
+	                    nargs='?')
+
+
+	args = parser.parse_args()
+
+	try:
+		to_host = args.to_host if args.to_host else os.environ["TO_URL"]
+	except KeyError as e:
+		raise KeyError("Traffic Ops hostname not set! Set the TO_URL environment variable or use "\
+		               "'--to_url'.") from e
+
+	useSSL = True
+	to_port = 443
+	if to_host.lower().startswith("http://"):
+		to_host = to_host[7:]
+		useSSL = False
+		to_port = 80
+	elif to_host.lower().startswith("https://"):
+		to_host = to_host[8:]
+
+	portpos = to_host.find(':')
+	if portpos > 0:
+		to_port = int(to_host[portpos+1:])
+		to_host = to_host[:portpos]
+
+	s = TOSession(to_host,
+	              host_port=to_port,
+	              ssl=useSSL,
+	              api_version=str(args.api_version),
+	              verify_cert=not args.insecure)
+	# s.create()
+
+	# The TOSession object's methods will handle '/' stripping for us, so it's only necessary with
+	# raw paths.
+	path = args.PATH if not args.raw_path else '/'.join((s.to_url.rstrip('/'), args.PATH.lstrip('/')))
+
+	data = args.DATA
+	if data and os.path.isfile(data):
+		with open(data) as f:
+			data = f.read()
+
+	if isinstance(data, str):
+		data = data.encode()
+
+	# if os.path.isfile(COOKIEFILE):
```

Comment:
This comment should be removed. Non-implemented functionality should not be commented in the code.

Reply:
gone